File: server/app/api/hub.py
"""xAppHub endpoints. Owner: Person 4 | BL-HUB-01."""

from fastapi import APIRouter

from app.schemas.hub import (
    AccessControlRequest,
    AccessControlResponse,
    AppCatalogItem,
    HubAnalytics,
)
from app.services import hub_service

router = APIRouter(prefix="/api/v1", tags=["hub"])


@router.get("/projects", response_model=list[AppCatalogItem])
async def catalog() -> list[AppCatalogItem]:
    return await hub_service.catalog()


@router.post("/hub/access", response_model=AccessControlResponse)
async def set_access(req: AccessControlRequest) -> AccessControlResponse:
    return await hub_service.set_access(req.project_id, req.user_email, req.role)


@router.get("/hub/analytics", response_model=HubAnalytics)
async def analytics() -> HubAnalytics:
    return await hub_service.analytics()

File: client/src/api/hub.js
/** xAppHub API - contract: server/app/schemas/hub.py (BE: Person 4, FE: Person 2). */
import { isDemoMode } from '@/config';
import catalogMock from '@/mocks/hub_catalog_mock.json';
import analyticsMock from '@/mocks/hub_analytics_mock.json';
import { http } from './http';
import { demo } from './demo';

export const hubApi = {
  catalog() {
    if (isDemoMode) return demo(catalogMock);
    return http.get('/api/v1/projects');
  },

  setAccess({ projectId, userEmail, role }) {
    if (isDemoMode) {
      return demo({ project_id: projectId, user_email: userEmail, role, updated: true });
    }
    return http.post('/api/v1/hub/access', { project_id: projectId, user_email: userEmail, role });
  },

  analytics() {
    if (isDemoMode) return demo(analyticsMock);
    return http.get('/api/v1/hub/analytics');
  },
};
